### PR TITLE
Ditaa extension for AsciidocFX

### PR DESCRIPTION
**Pull request by joemat**
Hi, 

I added an extension for creating ditaa diagrams (see http://ditaa.sf.net) with AsciidocFX.

Since the ditaa classes are already part of the plantuml package this doesn't need any additional dependencies.

I'd be happy if you'd merge this into the official source,

Kind regards 
Jörg

### CODE REVIEW

File: conf/booksample/chapter-05.adoc [hunk 1]
```diff
@@ -1,48 +1,17 @@
-== Using Mathematical Formulas
```
Comment (rahmanusta) [inline conf/booksample/chapter-05.adoc]: @joemat  Why did you remove this ? :)

File: src/main/java/com/kodcu/service/extension/DitaaService.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+		if (!fileName.endsWith(".png"))
+			return;
+
+		Integer cacheHit = current.getCache().get(fileName);
+		int hashCode = (fileName + ditaa).hashCode();
+		if (Objects.nonNull(cacheHit))
+			if (hashCode == cacheHit)
+				return;
+
+		threadService.runTaskLater(() -> {
```
Comment (rahmanusta) [inline src/main/java/com/kodcu/service/extension/DitaaService.java]: It seems current line running on worker Thread (https://github.com/asciidocfx/AsciidocFX/pull/121/files#diff-a81dffb1673c900f6b30da5f71cb984fR1688), for this reason we dont need to run on another worker Thread again.

**Comment by rahmanusta** [comment]
Hello @joemat, I'm happy for your contribution to AsciidocFX with ditaa extension. I commented some section, can you rebase those 3 commits to 1 commit ?

File: conf/booksample/chapter-05.adoc (hunk 1 above)
Comment (joemat) [inline conf/booksample/chapter-05.adoc]: It just looks as if - I tought it would be nice to have the docu for ditaa near the plantuml docu and therefor moved the chapters. I changed this in the new version and added the ditaa docu as `chapter-08.adoc`.

File: src/main/java/com/kodcu/service/extension/DitaaService.java (hunk 2 above)
Comment (joemat) [inline src/main/java/com/kodcu/service/extension/DitaaService.java]: To be honest: I copied the code from the plantuml extension and haven't thought about if it makes sense. Thank your for reviewing it, I changed it in the new version.

**Comment by joemat** [comment]
@rahmanusta, thanks for your reply and the code review. I fixed the issues you found, please come back to me if there are additional issues.

**Comment by wfouche** [comment]
Ditaa is already supported. I've been using it for a while. This Java extension is not required. In the current version just type the following to get ditaa support.

.UML http://plantuml.com/ditaa.html
[uml,file="ditaa-diagram.png"]
--
@startditaa
+--------+   +-------+    +-------+
|        +---+ ditaa +--> |       |
|  Text  |   +-------+    |diagram|
|Document|   |!magic!|    |       |
|     {d}|   |       |    |       |
+---+----+   +-------+    +-------+
    :                         ^
    |       Lots of work      |
    +-------------------------+
@endditaa
--

Works like charm.

**Comment by rahmanusta** [comment]
@joemat's pull request learnt me plantuml supports also ditaa. After understanding that, as @wfouche 's example @startdiaa @endditaa tags helps to render ditaa. @joemat Can you try that?

**Comment by joemat** [comment]
I didn't know that I can create ditaa diagrams with plantuml . .that works fine and solves my root issue.

Nevertheless I think it would be a good thing to have direct support for ditaa diagrams in AsciidocFX to make it compatible to [Asciidoctor diagrams](http://asciidoctor.org/docs/asciidoctor-diagram/).

It rebased my fork and changed the logic to use the @startditaa and @endditaa tags since this simplifies the implementation.

**Comment by rahmanusta** [comment]
Thanks for you contribution @joemat !

We are thinking to support/switch-to Asciidoctor diagrams format but there is a problem that is making me thoughtful.

For example, in the following ditaa block, when you change the content, for every keypress hashcode of content will changed, and it will affect the filename. And you will see 1 image file for 1 change. I dont know how to handle this scenario? There is no problem for Asciidoctor, because you are writing and generating the outputs. But, in AsciidocFX we are generating outputs in live mode.

[ditaa]
....
//
....
